src/sidebar/services/api.js

```diff
@@ -94,6 +94,32 @@ function serializeParams(params) {
   return parts.join('&');
 }
 
+/**
+ * @typedef APIResponse
+ * @prop {any} data - The JSON response from the API call.
+ * @prop {string|null} token - The access token that was used to make the call
+ *   or `null` if unauthenticated.
+ */
+
+/**
+ * Options controlling how an API call is made or processed.
+ *
+ * @typedef APICallOptions
+ * @prop [boolean] includeMetadata - If false (the default), the response is
+ *   just the JSON response from the API. If true, the response is an `APIResponse`
+ *   containing additional metadata about the request and response.
+ */
+
+/**
+ * Function which makes an API request.
+ *
+ * @typedef {function} APICallFunction
+ * @param [any] params - A map of URL and query string parameters to include with the request.
+ * @param [any] data - The body of the request.
+ * @param [APICallOptions] options
+ * @return {Promise<any|APIResponse>}
```

Comment:
Why return just a `Promise<APIResponse>`? Because that would involve changing all existing places that make API calls. Returning a union here avoids the need to do that, at least in this PR.

Reply:
Right!